File: src/suikailauncher/core/mod/modrinth.py
import pathlib
import sys
from suikailauncher.core.base import network
from suikailauncher.core.base.logger import logger
from suikailauncher.core.base import to_json,system

modrinth_api = "https://api.modrinth.com/v2"

modrinth_mirror_api = "https://mod.mcimirror.top/modrinth/v2"

# [["categories:forge"],["versions:1.17.1"],["project_type:mod"]]

async def search_mod(mod_name:str,mod_loader:str = "",version:str = ""):
    search_url = modrinth_api+f"/search?query={mod_name}&limit=50&facets=[[\"project_type:mod\"]"
    if mod_loader:
        search_url += f"[\"categories:{mod_loader.lower()}\"]"
    if version:
        search_url += f"[\"version:{version}\"]"
    search_url += "]"
    mirror_api = search_url.replace(modrinth_api,modrinth_mirror_api)
    search_result = await network.network_request(search_url)

    
    result = search_result.json()
    mods = result.get("hits")
    if not mods:
        return []
    for mod in mods:
        logger.info("")

async def get_project_meta(project_id_or_slug:str):
    pass

async def get_meta_by_hash(files:list[pathlib.Path]):
    hashes = []
    # 计算文件哈希（SHA 256）
    for file in files:
        hashes.append(system.get_file_or_string_hash("sha256",file=file.absolute()))
    # 构建请求数据
    data = {
        "hashes": hashes,
        "algorithm": "sha256"
    }
    modrinth_resp = await network.network_request(modrinth_api + "/version_files", "POST",{"Content-Type":"application/json"}, data=to_json(data))
    if modrinth_resp.is_error():
        return {}
    return modrinth_resp.json()
    

File: SuikaiLauncher.Core.Mod/Modules/Modrinth.cs
using System.ComponentModel.DataAnnotations;
using SuikaiLauncher.Core.Base;
using SuikaiLauncher.Core.Mod.JsonModels;
using SuikaiLauncher.Core.Override;

namespace SuikaiLauncher.Core.Mod{

    public class Modrinth
    {

        private static string ModrinthBaseAPI = "https://api.modrinth.com/v2";
        public static async Task<ModrinthSearchResult> SearchMod(string ModName, string? ModLoader, string? McVersion)
        {
            string SearchAPI = ModrinthBaseAPI + $"/search?query={ModName}&limit=100&facets=[[\"project_type:mod\"]";
            if (!ModLoader.IsNullOrWhiteSpaceF()) SearchAPI += $"[\"categories:{ModLoader}\"";
            if (!McVersion.IsNullOrWhiteSpaceF()) SearchAPI += $"[\"versions:{McVersion}\"]";
            SearchAPI += "]";
            HttpResponseMessage SearchResult = await Network.NetworkRequest(SearchAPI);
            ModrinthSearchResult Result = Json.GetJson<ModrinthSearchResult>(await FileIO.ReadAsString(SearchResult));
            int SearchCount = 0;
            foreach (var Project in Result.hits)
            {
                if (Project is null) continue;
                Project.ModI18nzhName = ModData.GetModTranslateName(Project.slug);
                Result.hits[SearchCount] = Project;
            }
            return Result;
        }
        
    }
}